src/history.cpp

```diff
@@ -1778,6 +1800,25 @@ void history_sanity_check() {
     // No sanity checking implemented yet...
 }
 
+wcstring history_session_id() {
+    wcstring result;
+
+    const env_var_t session_id = env_get_string(L"FISH_HISTFILE");
+    if (session_id.missing() || session_id.compare(L"default") == 0) {
+        result = L"fish";
+    } else if (session_id.empty()) {
+        result = L"";
+    } else if (valid_var_name(session_id)) {
+        result = session_id;
+    } else {
+        // TODO: the session id is invalid. Print an error.
+		debug(1, "TODO ERROR: Wrong characters in the session ID: '%ls'. Falling back to `fish`.", session_id.c_str());
```

Comment:
Why not just "History session ID '%ls' is not a valid var name. Falling back to `fish`."? Also, this should be localized by wrapping it in the magic `_()` function.